packages/fx-core/src/component/generator/apiSpec/helper.ts

```diff
@@ -649,21 +649,24 @@ export async function injectAuthAction(
     return res;
   } else if (
     (!!authScheme && Utils.isOAuthWithAuthCodeFlow(authScheme)) ||
-    authType === "OAuth2PluginVault"
+    authType === "OAuthPluginVault" ||
+    authType === "MicrosoftEntra"
   ) {
     const res = await ActionInjector.injectCreateOAuthAction(
       ymlPath,
       authName,
       relativeSpecPath,
-      forceToAddNew
+      forceToAddNew,
+      authType === "MicrosoftEntra"
     );
 
     if (await fs.pathExists(localYamlPath)) {
       await ActionInjector.injectCreateOAuthAction(
         localYamlPath,
         authName,
         relativeSpecPath,
-        forceToAddNew
+        forceToAddNew,
+        authType === "MicrosoftEntra"
```

Comment:
Since "MicrosoftEntra" is used in many places, it is better to use a constant to replace it